packages/types/message.d.ts

```diff
@@ -0,0 +1,24 @@
+export type CumulusMeta = {
+  execution_name: string
+  queueName?: string
+  state_machine: string
+  parentExecutionArn?: string
+  asyncOperationId?: string
+};
+
+export type ReplaceConfig = {
+  Bucket: string
+  Key: string
+  TargetPath?: string
+};
+
+export type CumulusMessage = {
+  cumulus_meta: CumulusMeta
+  meta: object
+  payload: unknown
+};
+
+export type CumulusRemoteMessage = {
```

Comment:
Looking at these docs:

[Depending on configuration other portions of the message may be present](https://github.com/nasa/cumulus/blob/4fd519d75513058f592c05abe4177245e70ad9ec/docs/workflows/cumulus-task-message-flow.md#L52)

This type is going to fail if one of those optional properties _is_ present (`payload`, for example). We should probably add `meta`, `payload`, and `exception` as optional on here.

Reply:
Is there still value in having this separate type at that point?

Reply:
I think there is. The "Cumulus Message" that most of our code is going to be working with will be guaranteed to have "meta", "cumulus_meta", "payload", and not "replace".